### PR TITLE
Retrun errors to client

### PR DESCRIPTION
**Pull request by pouladpld**
Based on the [GraphQL Schema](http://facebook.github.io/graphql/October2016/#sec-Errors), errors that happen during resolving the data should be returned to the client.

Currently, exception is thrown and client gets a 500 status code.

### CODE REVIEW

File: GraphQl.AspNetCore/GraphQlMiddleware.cs [hunk 1]
```diff
@@ -61,27 +59,9 @@ public class GraphQlMiddleware
         {
             var json = new DocumentWriter(indent: _options.FormatOutput).Write(result);
 
-            httpContext.Response.StatusCode = 200;
+            httpContext.Response.StatusCode = result.Errors?.Any() == true ? 400 : 200;
```
Comment (pouladpld) [inline GraphQl.AspNetCore/GraphQlMiddleware.cs]: I changed my mind about status codes and decided to always respond with 200 OK to client. GraphQL has nothing to do with HTTP standards. It has its own way to tell clients whether execution was good or there were problems.

**Comment by JuergenGutsch** [comment]
Hi @pouladpld,

Thanks again for your PR :-)

Would you please use the develop branch as your base for PRs? Since I use git flow, the working branch is **develop** and the production branch is master. I wrote about that in the second paragraph in the [README.md](https://github.com/JuergenGutsch/graphql-aspnetcore/blob/master/README.md). Also the **develop** branch contains some more fixes and improvements, which are not yet in master.

I'm sure this is solved in a different way in the develop branch. We changed a lot in this version. I'm going to close this PR because of that.

**Comment by pouladpld** [comment]
@JuergenGutsch Yes. The code on `develop` has this fix. Could you release another pre-release?

**Comment by JuergenGutsch** [comment]
Hey @pouladpld, the merge of your other PR just triggered a [successful build](https://ci.appveyor.com/project/JuergenGutsch/graphql-aspnetcore/branch/develop) and it did a [new pre-release on MyGet](https://www.myget.org/feed/Packages/juergengutsch).

**Pull request closed without merging**